doc/guides/maintaining-npm.md

```diff
@@ -0,0 +1,66 @@
+# Maintaining npm in Node.js
+
+## Step 1: Clone npm
+
+```console
+$ git clone https://github.com/npm/npm.git
+$ cd npm
+```
+
+or if you already have npm cloned make sure the repo is up to date
+
+```console
+$ git remote update -p
+$ git reset --hard origin master
+```
+
+## Step 2: Build release
+
+```console
+$ git checkout vX.Y.Z
+$ make release
+```
+
+## Step 3: Remove old npm
+
+```console    
+$ cd /path/to/node
+$ git checkout master
+$ git remote update -p
+$ git reset --hard origin master
+$ git checkout -b npm-x.y.z
+$ cd deps
+$ rm -rf npm
+```
+
+## Step 4: Extract and commit new npm
+
+```console
+$ tar zxf /path/to/npm/release/npm-x.y.z.tgz
+$ git add -f npm
```

Comment:
Shouldn't this be `-A`, to make sure files are both added and removed?

Reply:
When I was doing it with `-A` it wasn't adding all the files. I believe `add -f` will add and remove.

Reply:
As of git 2.0, `git add` [adds removed files as well as added files](https://github.com/git/git/blob/14c63a9dc093d6738454f6369a4f5663ca732cf7/Documentation/RelNotes/2.0.0.txt#L32-L36).

`git add -f` is about adding `gitignore`d files, not sure we'd want that.

Reply:
Yeah, if some files are missing, we should probably modify a rule in our `.gitignore`

Reply:
I've updated and opened https://github.com/nodejs/node/pull/17363